py-rse/teams.Rmd

```diff
@@ -318,91 +366,70 @@ Every issue can hold:
-and should be enforced by the project's Code of Conduct (Section \@ref(#teams-coc)).
+> Some projects go so far as to create a page of links
+> to old questions and answers that are particularly helpful.
 
-An informative bug report should include each of the following,
- to the best of your ability.
+## Bug Reports {#teams-bugs}
+
+A well-written bug report is more likely to get a fast response,
+and is more likely to get a response actually addresses the issue @Bett2008.
+To write a good bug report:
 
 1.  Make sure the problem actually *is* a bug.
     It's always possible that we have called a function the wrong way
-    or done an analysis using the wrong configuration file;
-    if we take a minute to double-check,
+    or done an analysis using the wrong configuration file.
+    If we take a minute to double-check,
+    or ask someone else on our team to check our logic,
     we could well fix the problem ourselves.
 
-2.  Try to come up with a [reproducible example][reprex] or "reprex".
-    A reprex includes only the steps or lines of code needed to make the problem happen;
-    again,
-    it's surprising how often we can solve the problem ourselves
-    as we trim down the steps to create one.
+2.  Try to come up with a [reproducible example][reprex] or "reprex"
+    that includes only the steps or lines of code needed to make the problem happen.
+    Again,
+    we can often we solve the problem ourselves as we trim down the steps to create one.
 
 3.  Write a one-line title for the issue
-    and a longer (but still brief) description that includes relevant details).
+    and a longer (but still brief) description that includes relevant details.
 
-4.  Attach any screenshots that show the problem, resulting errors, or (slimmed-down) input files needed to re-create it.
+4.  Attach any screenshots that show the problem,
+    resulting errors,
+    or (slimmed-down) input files needed to re-create it.
 
 5.  Describe the version of the software we were using,
-    the operating system we were running on, which version of the programming language you ran it with,
-    and anything else you believe that might affect behavior.
+    the operating system we were running on,
+    which version of the programming language we ran it with,
+    and anything else that might affect behavior.
+    If the software in question uses a logging framework (Section \@ref(errors-logging)),
+    turn debugging output on and include it with the issue.
 
 6.  Describe each problem separately so that each one can be tackled on its own.
     This parallels the rule about creating a branch in version control for each bug fix or feature
     discussed in Section \@ref(git-advanced).
 
-Here is an example of a well-written bug report with all of the components mentioned above,
-as well as the common features of all issues:
+Here is an example of a well-written bug report with all of the components mentioned above:
 
 ```text
 ID: #25
```

Comment:
The creator and owner in his example bug report could be Sami and Amira to fit with our narrative?

Also, the current example bug report uses a hypothetical script `wordbase.py`, which is somewhat confusing because it sounds like one of the Zipf's scripts (`countwords.py`) but isn't. It might be better to submit a genuine bug report? For instance, `countwords.py` doesn't handle em-dashes properly. Instead of splitting on an em-dash, it counts the whole thing as a word. e.g.
```
$ python bin/countwords.py data/dracula.txt
...
time--which,1
sun--now,1
tops--and,1
off--in,1
before--came,1
... 
```
(That's just a few of the many em-dash outputs when you run `countwords.py` on Dracula.)

Reply:
Once we've created a `zipf` directory in Amira's repo we could also put the text from this bug report into an actual issue at that repo and screenshot it?
https://github.com/amirakhan

Reply:
1. Changed bug report - good catch.
2. Filed issue #405 about screenshot.